charts/internal/calico/templates/node/daemonset-calico-node.yaml

```diff
@@ -101,11 +101,23 @@ spec:
         # host.
         - name: calico-node
           image: {{ index .Values.images "calico-node" }}
+          ports:
+            {{- if .Values.config.monitoring.enabled }}
```

Comment:
Out of curiosity: What is the rationale of making this monitoring configurable? When would I want to disable it?

Reply:
if it causes unnecessary overhead on the seed Prometheus is one case, also for test clusters.

Reply:
> if it causes unnecessary overhead on the seed Prometheus is one case, also for test clusters.

Hm, ok, I'm asking because this flag is exposed to end-users. Hence, when - as an end-user - would I disable this? Probably never as "I don't know/care" about the Prometheus running in the seed, so I just always enable it.
If there are no other use-cases, wouldn't it make more sense to not expose it in the `NetworkConfig` but do it as you suggested - not enable it for `testing` purpose cluster, for example?

> One other thing, if we wish to change the ports in the future for whatever reason

Didn't get this one, can't we change them either way?

Reply:
> Didn't get this one, can't we change them either way?

We'd have to create a new release for that since we will be modifying the charts or? 

I see your point. Exposing it to the users does not bring any meaning, however, I am not sure if we want to tweak anything in the future related to monitoring, we currently don't have the distinction between an end-user or an operator when the shoot is modified. But if we do, then we can hide fields like this one, and expose it to operators only. 

That said, I have no strong opinion regarding disabling it.

Reply:
Hm, sounds theoretical to me. We don't have these needs today, and there is no additional configuration option yet for these monitoring things --> I wouldn't even bother to introduce this in the API and save same code, and rather revisit when it gets relevant again (if we don't have some use-cases today).
I kind of get the point with the `testing` clusters, although such clusters don't get a monitoring stack in the first place, so these configuration artefacts cannot put any load on the non-existing Prometheus.

To sum up, I would save this additional complexity+code (yes, it's not very complex, but still) for now, but that's just my high level recommendation without being into the details too much.